crates/sncast/src/lib.rs

```diff
@@ -327,6 +328,42 @@ fn get_account_from_accounts_file<'a>(
     Ok(account)
 }
 
+pub async fn check_if_legacy_contract(
+    class_hash: Option<FieldElement>,
+    address: FieldElement,
+    provider: &JsonRpcClient<HttpTransport>,
+) -> Result<bool> {
+    let contract_class = match class_hash {
+        Some(class_hash) => provider.get_class(BlockId::Tag(Pending), class_hash).await,
+        None => provider.get_class_at(BlockId::Tag(Pending), address).await,
```

Comment:
should this use `get_class_hash_by_address`?

Reply:
We want to receive a contract class, not a class hash. I initially moved `get_class_hash_by_address` here since it was utilized in the first solution. If `account/add.rs` is a better location for it, it can certainly be moved back.

Reply:
ahh shit, misread. sorry.